src/main/drivers/accgyro/accgyro_spi_lsm6dsv16x.c

```diff
@@ -964,9 +967,30 @@ void lsm6dsv16xGyroInit(gyroDev_t *gyro)
                                     LSM6DSV_CTRL2_ODR_G_MASK,
                                     LSM6DSV_CTRL2_ODR_G_SHIFT));
 
+    switch (gyroConfig()->gyro_hardware_lpf) {
+        default:
+        case GYRO_HARDWARE_LPF_NORMAL:
+            gyro_lpf = LSM6DSV_CTRL6_FS_G_BW_407HZ;
+            break;
+
+        case GYRO_HARDWARE_LPF_OPTION_1:
+            gyro_lpf = LSM6DSV_CTRL6_FS_G_BW_281HZ;
+            break;
+
+        case GYRO_HARDWARE_LPF_OPTION_2:
+            gyro_lpf = LSM6DSV_CTRL6_FS_G_BW_213HZ;
+            break;
+
+#ifdef USE_GYRO_DLPF_EXPERIMENTAL
+        case GYRO_HARDWARE_LPF_EXPERIMENTAL:
+            gyro_lpf = LSM6DSV_CTRL6_FS_G_BW_156HZ;
+            break;
+#endif
+        }
+
```

Comment:
```suggestion
   static const uint8_t  lpf_map[GYRO_HARDWARE_LPF_COUNT] = {
        [GYRO_HARDWARE_LPF_NORMAL] = LSM6DSV_CTRL6_FS_G_BW_407HZ,
        [GYRO_HARDWARE_LPF_OPTION_1]= LSM6DSV_CTRL6_FS_G_BW_281HZ,
        [GYRO_HARDWARE_LPF_OPTION_2] = LSM6DSV_CTRL6_FS_G_BW_213HZ,
#ifdef USE_GYRO_DLPF_EXPERIMENTAL
        [GYRO_HARDWARE_LPF_EXPERIMENTAL] = LSM6DSV_CTRL6_FS_G_BW_156HZ,
#endif
        };

        const uint8_t gyro_lpf = (gyroConfig()->gyro_hardware_lpf < ARRAYLEN(gyro_lpf_map)) ? gyro_lpf_map[gyroConfig()->gyro_hardware_lp] : GYRO_HARDWARE_LPF_NORMAL;
```

Reply:
Admittedly cleaner although your final statement would assign the value `GYRO_HARDWARE_LPF_NORMAL` to `gyro_lpf` rather than a default filter setting. Also has a typo.

I used a switch as I anticipate potentially more complex changes per filter option. Let’s recode once the flight testers have determined good options.